app/src/main/java/com/example/antitheft/structure/LocationParser.java

```diff
@@ -0,0 +1,137 @@
+package com.example.antitheft.structure;
+
+import android.location.Location;
+
+import androidx.annotation.NonNull;
+
+/**
+ * This class retrieves the coordinates of a certain {@link android.location.Location} , and has
+ * a custom toString method.
+ *
+ * @author Francesco Bau'
+ * @version 0.1
+ * @since 26/02/2020
+ */
+public class LocationParser {
+    private static final String UNKNOWN_LOCATION_MESSAGE = "Location is NOT acquired.";
+    protected static final double DEFAULT_LOCATION = 0;
+    static final String DEFAULT_PROVIDER = "Default";
+
+    private double latitude, longitude;
+    private boolean defaultFlag;
```

Comment:
You are right.
Though at first, I thought that way; in fact, it was called `isAcquired()` , and the field was called `acquired`, but then I also realized that even mocked Locations can have a (mocked) accuracy...
So "acquired" is not appropriate, otherwise, the check would have been more complex.

Also, at first, there was another constructor, which was `LocationParser(double,double)` , and method `setLocation(double,double)` was public. But then I decided to make it accessible with a `Location` value only, due to possible security issues, and because it started to become hard to check if it was actually acquired or not.

In the end, I decided to rename it `defaultFlag`, because it was easier to handle.

I'm gonna write more comments, though.
It seems like it wasn't clarified enough.

Thank you!

Reply:
Yes, exactly, try to give a more meaningful name or explain it better.